docking.js

```diff
@@ -2105,6 +2136,7 @@ var DockManager = class DashToDock_DockManager {
     }
 
     _deleteDocks() {
+        this.desktopIconsUsableArea.resetMargins();
```

Comment:
I'd prefer this to be in `toggle()` instead.

Reply:
But they aren't "toggled", but "erased".

Reply:
Yeah, I know... But I'm saying with this to handle it in the caller function, as `_deleteDocks` is relevant to docks only. 

However, I'm wondering if we couldn't just handle it when destroying each dock, so basically ensure that `desktopIconsUsableArea.setMargins(this._monitorIndex, 0, 0, 0, 0);` is called on each dock's destroy().

Reply:
I wanted to keep the API as simple as possible. That's why each time you want to remove one element, you have to delete all and reinsert the remainder ones. Also, that's another reason why I used a timer instead of the idle.